pkg/proxy/userspace/proxysocket.go

```diff
@@ -162,11 +160,13 @@ func copyBytes(direction string, dest, src *net.TCPConn, wg *sync.WaitGroup) {
 	glog.V(4).Infof("Copying %s: %s -> %s", direction, src.RemoteAddr(), dest.RemoteAddr())
 	n, err := io.Copy(dest, src)
 	if err != nil {
-		glog.Errorf("I/O error: %v", err)
+		if !isClosedError(err) {
+			glog.Errorf("I/O error: %v", err)
+		}
 	}
 	glog.V(4).Infof("Copied %d bytes %s: %s -> %s", n, direction, src.RemoteAddr(), dest.RemoteAddr())
-	dest.CloseWrite()
-	src.CloseRead()
+	dest.Close()
+	src.Close()
```

Comment:
If you only close the destination you will end up closing both connections but without double closes and in turn without needing the `isClosedError()` check.